ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/encoding/TransactionEncoder.java

```diff
@@ -49,10 +49,16 @@ public static void encodeForWire(final Transaction transaction, final RLPOutput
     final TransactionType transactionType =
         checkNotNull(
             transaction.getType(), "Transaction type for %s was not specified.", transaction);
+    encodeForWire(transactionType, encodeOpaqueBytes(transaction), rlpOutput);
+  }
+
+  public static void encodeForWire(
+      final TransactionType transactionType, final Bytes opaqueBytes, final RLPOutput rlpOutput) {
+    checkNotNull(transactionType, "Transaction type was not specified.");
     if (TransactionType.FRONTIER.equals(transactionType)) {
-      encodeFrontier(transaction, rlpOutput);
+      rlpOutput.writeRaw(opaqueBytes);
     } else {
-      rlpOutput.writeBytes(encodeOpaqueBytes(transaction));
+      rlpOutput.writeBytes(opaqueBytes);
```

Comment:
I scrutinized this to be sure we are not creating a regression for non-frontier transactions.  LGTM